salt/etc-hosts/hosts.jinja

```diff
@@ -1,5 +1,7 @@
 ### service names ###
+{% if "kube-master" in salt['grains.get']('roles', []) %}
```

Comment:
As far as I know, the API server is configured in 127.0.0.1 in all the machines in the cluster because we have haproxy listening in localhost and it will forward connections to the real API server. So I think this `127.0.0.1 api` entry should be present in all the machines...

Reply:
I think we should remove the last part of the entry, so it should just be `127.0.0.1 api api.{{ pillar['internal_infra_domain'] }}` (but my guts tell me there will be some reason for this related to certificates). Maybe @kiall knows about this...

Reply:
I think we could only include the FQDN alias in the master by using something like:

```language:SaltStack
127.0.0.1 api api.{{ pillar['internal_infra_domain'] }}
{%- if "kube-master" in salt['grains.get']('roles', []) -%}
{{- pillar['api']['server']['external_fqdn'] -}} {#- this should be appended to the previous line -#}
{%- endif %}
```
So we would get something like `127.0.0.1 api api.infra.caasp.local c166.arch.suse.de` in the master but `127.0.0.1 api api.infra.caasp.local` in the rest of the cluster.